src/System.Net.Http/src/System/Net/Http/Managed/HttpSystemProxy.cs

```diff
@@ -0,0 +1,194 @@
+        private bool _bypassLocal = false;      // we should bypass domain considered local
+        private ICredentials _credentials;
+        private readonly WinInetProxyHelper _proxyHelper;
+        private SafeWinHttpHandle _sessionHandle;
+        private volatile bool _disposed;
+
+        public static bool TryCreate(out IWebProxy proxy)
+        {
+            // This will get basic proxy setting from system using existing
+            // WinInetProxyHelper functions. If no proxy is enabled, it will return null.
+            SafeWinHttpHandle sessionHandle = null;
+            proxy = null;
+
+            WinInetProxyHelper proxyHelper = new WinInetProxyHelper();
+            if (!proxyHelper.ManualSettingsOnly && !proxyHelper.AutoSettingsUsed)
+            {
+                return false;
+            }
+            if (proxyHelper.AutoSettingsUsed)
+            {
+                sessionHandle = Interop.WinHttp.WinHttpOpen(
+                    IntPtr.Zero,
+                    Interop.WinHttp.WINHTTP_ACCESS_TYPE_NO_PROXY,
+                    Interop.WinHttp.WINHTTP_NO_PROXY_NAME,
+                    Interop.WinHttp.WINHTTP_NO_PROXY_BYPASS,
+                    (int)Interop.WinHttp.WINHTTP_FLAG_ASYNC);
+                if (sessionHandle.IsInvalid)
+                {
+                    // Proxy failures are currently ignored by managed handler.
+                    return false;
+                }
+            }
+            proxy  = new HttpSystemProxy(proxyHelper, sessionHandle);
+            return true;
+        }
+
+        private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessionHandle)
+        {
+            _proxyHelper = proxyHelper;
+            _sessionHandle = sessionHandle;
+
+            if (proxyHelper.ManualSettingsOnly)
+            {
+                _proxyUri = GetUriFromString(proxyHelper.Proxy);
+
+                if (!string.IsNullOrWhiteSpace(proxyHelper.ProxyBypass))
+                {
+                    // Process bypass list for manual setting.
+                    string[] list = proxyHelper.ProxyBypass.Split(';');
+                    List<string> tmpList = new List<string>();
+
+                    foreach (string value in list)
+                    {
+                        string tmp = value.Trim();
+                        if (tmp == "<local>")
+                        {
+                            _bypassLocal = true;
+                            continue;
+                        }
+                        if (tmp.Length > 0)
+                        {
+                            tmpList.Add(tmp);
+                        }
+                    }
+                    if (tmpList.Count > 0)
+                    {
+                        _bypass = tmpList.ToArray();
+                    }
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        internal void Dispose(bool disposing)
```

Comment:
This type doesn't have (or need) a finalizer, so Dispose will never be called with false, and you can just consolidate this all into the single public void Dispose() method.